DistroLauncher/named_mutex.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "stdafx.h"
+
+DWORD Win32MutexApi::create(HANDLE& mutex_handle, LPCWSTR mutex_name) noexcept
+{
+    mutex_handle = CreateMutex(nullptr, FALSE, mutex_name);
+    if (mutex_handle == nullptr) {
```

Comment:
Just a side comment: This is correct, but some Win32 functions returns `INVALID_HANDLE_VALUE` (`#define INVALID_HANDLE_VALUE ((HANDLE)(LONG_PTR)-1)`) instead of `NULL`, so make sure to always check the docs, because that value compares false against `nullptr`.

Reply:
Good catch

Reply:
I re-read the documentation and this function in particular always returns NULL when failed:
> If the function succeeds, the return value is a handle to the newly created mutex object.
> If the function fails, the return value is NULL. To get extended error information, call [GetLastError](https://docs.microsoft.com/en-us/windows/desktop/api/errhandlingapi/nf-errhandlingapi-getlasterror).

THanks anyways for the heads-up because I did not know.